one-line-cr-bot.sh

```diff
@@ -0,0 +1,531 @@
+        echo "$_PROG: cleanup_handler() invoked, exit status $ERR" 1>&2
+
+        # Print collected error output in case we failed
+        echo "Full stderr:"
+        [ -d "$TMP_DATA_FOLDER" ] && cat "$TMP_DATA_FOLDER"/stderr.log 2>/dev/null | awk '{print "stderr.log:" $0}'
+    fi
+
+    [ -d "$TMP_DATA_FOLDER" ] && rm -rf "$TMP_DATA_FOLDER"
+
+    # Try to get back to the initial commit as before (might result in head-less
+    # state). This is to not confuse interactive users of this script.
+    if [ -n "$WORK_COMMIT" ] && [ -n "$BASE_COMMIT" ]; then
+        pushd "$REPOSITORY_DIR" 1>&2
+        git checkout "$WORK_COMMIT" &>/dev/null
+        git submodule update --init --recursive &>/dev/null
+        popd 1>&2
+    fi
+
+    return "$ERR"
+}
+
+error_handler() {
+    local _PROG="$0"
+    local LINE="$1"
+    local ERR="$2"
+    if [ "$ERR" != 0 ]; then
+        echo "$_PROG: error_handler() invoked, line $LINE, exit status $ERR" 1>&2
+    fi
+
+    exit "$ERR"
+}
+
+trap 'error_handler ${LINENO} $?' ERR
+trap 'cleanup_handler $?' EXIT
+
+check_environment() {
+    local -i ret=0
+    # Print the version output of the available tools
+    echo "Versions of used tools:" 1>&2
+    for tool in "${SUPPORTED_TOOLS[@]}"; do
+        if ! command -v "$tool" &>/dev/null && [ "${RUN_TOOL["${tool^^}"]}" == "true" ]; then
+            echo "Error: Failed to find tool $tool, which has been activated. Abort" 1>&2
+            ret=1
+        else
+            $tool --version 1>&2 || true
+        fi
+    done
+
+    return $ret
+}
+
+setup_infer() {
+
+    local INSTALL_DIR="$1"
+    pushd "$INSTALL_DIR" 1>&2
+
+    echo "Setting up Infer..." 1>&2
+
+    # prepare for checking downloaded tar ball (taken from release web page)
+    echo "510eeccc7e6bcc2678ac92a88f8e1cb9c07c3e14d272dcc06834e93845bb120f  infer-linux64-v1.0.0.tar.xz" >infer-linux64-v1.0.0.tar.xz.sha256
+
+    # get infer with specified version, check and install
+    local -i ret=0
+    curl -sSL "https://github.com/facebook/infer/releases/download/v${INFER_VERSION}/infer-linux64-v${INFER_VERSION}.tar.xz" \
+        --output "infer-linux64-v${INFER_VERSION}.tar.xz" || ret=$?
+
+    # fail in case the download failed
+    if [ "$ret" -ne 0 ]; then
+        popd 1>&2
+        return 1
+    fi
+
+    # check obtained artefact, extract and make available in PATH
+    if ! shasum --status --algorithm 256 --check infer-linux64-v1.0.0.tar.xz.sha256; then
+        echo "Error: checksum of obtained infer version does not match"
+        popd 1>&2
+        return 1
+    fi
+
+    tar -xJf "infer-linux64-v${INFER_VERSION}.tar.xz" || ret=$?
+    export PATH=$PATH:"$(pwd)/infer-linux64-v${INFER_VERSION}/bin"
+
+    popd 1>&2
+
+    return "$ret"
+}
+
+setup_cppcheck() {
+
+    local -i ret=0
+    local INSTALL_DIR="$1"
+    pushd "$INSTALL_DIR" 1>&2
+
+    echo "Setting up CppCheck..." 1>&2
+
+    # get cppcheck source
+    git --depth 1 -b 2.2 https://github.com/danmar/cppcheck.git || ret=$?
```

Comment:
In keeping with the approach for Infer, I'd go with downloading `https://github.com/danmar/cppcheck/archive/2.2.tar.gz` (and verify the checksum).

Reply:
Fair point, will adapt.